angr-management: init at 9.2.130

spencerpogo opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

https://github.com/angr/angr-management

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

spencerpogo committed:
python3Packages.binsync: init at 5.2.0

spencerpogo committed:
python3Packages.pyqodeng: init at 0.0.11

spencerpogo committed:
angr-management: init at 9.2.130

spencerpogo commented:
```
--------- Impacted packages on 'x86_64-linux' ---------
17 packages added:
angr-management (init at 9.2.130) python311Packages.binsync (init at 5.2.0) python311Packages.cmake-build-extension (init at 0.6.1) python311Packages.ghidra-bri
dge (init at 1.0.0) python311Packages.jfx-bridge (init at 1.0.0) python311Packages.libbs (init at 2.8.0) python311Packages.pyhidra (init at 1.3.0) python311Pack
ages.pyqodeng (init at 0.0.11) python311Packages.pyside6-qtads (init at 4.3.1.1) python312Packages.binsync (init at 5.2.0) python312Packages.cmake-build-extensi
on (init at 0.6.1) python312Packages.ghidra-bridge (init at 1.0.0) python312Packages.jfx-bridge (init at 1.0.0) python312Packages.libbs (init at 2.8.0) python31
2Packages.pyhidra (init at 1.3.0) python312Packages.pyqodeng (init at 0.0.11) python312Packages.pyside6-qtads (init at 4.3.1.1)

6 packages updated:
python3.11-angr python3.11-angrcli python3.11-angrop python3.12-angr python3.12-angrcli python3.12-angrop

--------- Report for 'x86_64-linux' ---------
46 packages built:
angr-management angr-management.dist python311Packages.angr python311Packages.angr.dist python311Packages.angrcli python311Packages.angrcli.dist python311Packages.angrop python311Packages.angrop.dist python311Packages.binsync python311Packages.binsync.dist python311Packages.cmake-build-extension python311Packages.cmake-build-extension.dist python311Packages.ghidra-bridge python311Packages.ghidra-bridge.dist python311Packages.jfx-bridge python311Packages.jfx-bridge.dist python311Packages.libbs python311Packages.libbs.dist python311Packages.pyhidra python311Packages.pyhidra.dist python311Packages.pyqodeng python311Packages.pyqodeng.dist python311Packages.pyside6-qtads python311Packages.pyside6-qtads.dist python312Packages.angr python312Packages.angr.dist python312Packages.angrcli python312Packages.angrcli.dist python312Packages.angrop python312Packages.angrop.dist python312Packages.binsync python312Packages.binsync.dist python312Packages.cmake-build-extension python312Packages.cmake-build-extension.dist python312Packages.ghidra-bridge python312Packages.ghidra-bridge.dist python312Packages.jfx-bridge python312Packages.jfx-bridge.dist python312Packages.libbs python312Packages.libbs.dist python312Packages.pyhidra python312Packages.pyhidra.dist python312Packages.pyqodeng python312Packages.pyqodeng.dist python312Packages.pyside6-qtads python312Packages.pyside6-qtads.dist
```

spencerpogo commented:
@ofborg build angr-management

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4898

spencerpogo commented:
- Fixed merge conflicts 
- Removing first-time contribution label (this is my first PR since I updated my username)

fabaff commented on pkgs/development/python-modules/angr/default.nix:
```diff
@@ -1,40 +1,45 @@
+    self = overriddenPython;
+    inherit packageOverrides;
+  };
+in
```
`angr` is a package and not an application. We agreed that overriding `unicorn` is the last resort as overriding module inside pythonPackages is a tricky thing because there are side effects.

The override should be a simple as possible (like the one we used in the [past](https://github.com/NixOS/nixpkgs/commit/e11030af6e7ea135291ebe151879bd1259e2ba73)).

fabaff commented on pkgs/development/python-modules/binsync/default.nix:
```diff
@@ -0,0 +1,62 @@
+
+  optional-dependencies = {
+    ghidra = [ pyside6 ];
+    test = [
+      flake8
+      pytest
+      pytest-qt
+    ];
```
```suggestion
    test = [
      flake8
      pytest
      pytest-qt
    ];
```

They should go into `nativeCheckInputs` and use `pytestCheckHook`. Usually we don't need the liniting tools.

fabaff commented on pkgs/development/python-modules/binsync/default.nix:
```diff
@@ -0,0 +1,62 @@
+
+  meta = {
+    description = "A reversing plugin for cross-decompiler collaboration, built on git";
+    homepage = "https://github.com/binsync/binsync";
```
```suggestion
    homepage = "https://github.com/binsync/binsync";
    changelog = "https://github.com/binsync/binsync/releases/tag/v${version}";
```

fabaff commented on pkgs/development/python-modules/binsync/default.nix:
```diff
@@ -0,0 +1,62 @@
+    description = "A reversing plugin for cross-decompiler collaboration, built on git";
+    homepage = "https://github.com/binsync/binsync";
+    license = lib.licenses.bsd2;
+    maintainers = with lib.maintainers; [ ];
```
New packages need a maintainer.
```suggestion
    maintainers = with lib.maintainers; [ ];
```

fabaff commented on pkgs/development/python-modules/cmake-build-extension/default.nix:
```diff
@@ -0,0 +1,51 @@
+    python.pkgs.gitpython
+  ];
+
+  # don't run cmake's configure phase
+  configurePhase = ''
+    runHook preConfigure
+    runHook postConfigure
+  '';
```
At first glance this could be replaced by `dontUseCmakeConfigure`.

fabaff commented on pkgs/development/python-modules/cmake-build-extension/default.nix:
```diff
@@ -0,0 +1,51 @@
+  dependencies = [
+    python.pkgs.cmake
+    python.pkgs.ninja
+    python.pkgs.gitpython
```
This looks wrong.

fabaff commented on pkgs/development/python-modules/cmake-build-extension/default.nix:
```diff
@@ -0,0 +1,51 @@
+    description = "Setuptools extension to build and package CMake projects";
+    homepage = "https://github.com/diegoferigo/cmake-build-extension";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ ];
```
See above

fabaff commented on pkgs/development/python-modules/cmake-build-extension/default.nix:
```diff
@@ -0,0 +1,51 @@
+  build-system = [
+    setuptools
+    setuptools-scm
+    wheel
```
`wheel` is usually not needed.

fabaff commented on pkgs/development/python-modules/cmake-build-extension/default.nix:
```diff
@@ -0,0 +1,51 @@
+
+  meta = {
+    description = "Setuptools extension to build and package CMake projects";
+    homepage = "https://github.com/diegoferigo/cmake-build-extension";
```
```suggestion
    homepage = "https://github.com/diegoferigo/cmake-build-extension";
    changelog = "https://github.com/diegoferigo/cmake-build-extension/releases/tag/v${version}";
```

fabaff commented on pkgs/development/python-modules/jfx-bridge/default.nix:
```diff
@@ -0,0 +1,39 @@
+    setuptools
+    wheel
+  ];
+
```
Upstream has tests (https://github.com/justfoxing/jfx_bridge/tree/master/jfx_bridge).

fabaff reviewed (commented):
Some comments apply to file module.

SuperSandro2000 commented on pkgs/development/python-modules/binsync/default.nix:
```diff
@@ -0,0 +1,62 @@
+  src = fetchFromGitHub {
+    owner = "binsync";
+    repo = "binsync";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

SuperSandro2000 commented on pkgs/development/python-modules/ghidra-bridge/default.nix:
```diff
@@ -0,0 +1,42 @@
+
+  build-system = [
+    setuptools
+    wheel
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/jfx-bridge/default.nix:
```diff
@@ -0,0 +1,39 @@
+  src = fetchFromGitHub {
+    owner = "justfoxing";
+    repo = "jfx_bridge";
+    rev = version;
```
```suggestion
    tag = version;
```

SuperSandro2000 commented on pkgs/development/python-modules/libbs/default.nix:
```diff
@@ -0,0 +1,57 @@
+
+  build-system = [
+    setuptools
+    wheel
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/libbs/default.nix:
```diff
@@ -0,0 +1,57 @@
+  src = fetchFromGitHub {
+    owner = "binsync";
+    repo = "libbs";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

SuperSandro2000 commented on pkgs/development/python-modules/libbs/default.nix:
```diff
@@ -0,0 +1,57 @@
+  pythonImportsCheck = [ "libbs" ];
+
+  meta = {
+    description = "A library for writing plugins in any decompiler: includes API lifting, common data formatting, and GUI abstraction";
```
```suggestion
    description = "Library for writing plugins in any decompiler: includes API lifting, common data formatting, and GUI abstraction";
```

SuperSandro2000 commented on pkgs/development/python-modules/pyhidra/default.nix:
```diff
@@ -0,0 +1,38 @@
+  src = fetchFromGitHub {
+    owner = "dod-cyber-crime-center";
+    repo = "pyhidra";
+    rev = version;
```
```suggestion
    tag = version;
```

SuperSandro2000 commented on pkgs/development/python-modules/pyqodeng/default.nix:
```diff
@@ -0,0 +1,48 @@
+  src = fetchFromGitHub {
+    owner = "angr";
+    repo = "pyqodeng";
+    rev = "v${version}";
```
```suggestion
    tag = "v${version}";
```

SuperSandro2000 commented on pkgs/development/python-modules/pyqodeng/default.nix:
```diff
@@ -0,0 +1,48 @@
+
+  build-system = [
+    setuptools
+    wheel
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/pyqodeng/default.nix:
```diff
@@ -0,0 +1,48 @@
+  pythonImportsCheck = [ "pyqodeng" ];
+
+  meta = {
+    description = "Angr's fork of pyQode.core, used as part of angr management";
```
Shouldn't the package attr reflect that?

spencerpogo replied:
I'm not sure what you're thinking of.

SuperSandro2000 commented on pkgs/development/python-modules/pyqodeng/relax-deps.patch:
Please use postPatch with susbtituteInPlace to not have to rebase the patch on every dependency change

SuperSandro2000 commented on pkgs/development/python-modules/pyside6-qtads/default.nix:
```diff
@@ -0,0 +1,75 @@
+    pyside6
+  ];
+  build-system = [
+    # cmake
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/pyside6-qtads/default.nix:
```diff
@@ -0,0 +1,75 @@
+    setuptools
+    setuptools-scm
+    shiboken6
+    wheel
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/pyside6-qtads/default.nix:
```diff
@@ -0,0 +1,75 @@
+    setuptools-scm
+    shiboken6
+    wheel
+    pyside6
+    shiboken6
```
Those are duplicated. Also are you sure they are needed here?

SuperSandro2000 commented on pkgs/development/python-modules/pyside6-qtads/default.nix:
```diff
@@ -0,0 +1,75 @@
+      --replace-fail @pyside6@ ${pyside6}
+  '';
+
+  buildInputs = [
+    qt6.qtbase
+    qt6.qtquick3d
+    shiboken6
+    pyside6
+  ];
```
```suggestion
  buildInputs = [
    qt6.qtbase
    qt6.qtquick3d
  ];
```

they are already in dependencies

SuperSandro2000 commented on pkgs/development/python-modules/pyside6-qtads/find-nix-deps.patch:
```diff
@@ -0,0 +1,58 @@
++    "PySide6",
+     "shiboken6",
+-    "shiboken6_generator"
++    # "shiboken6_generator"
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/python-modules/pyside6-qtads/find-nix-deps.patch:
```diff
@@ -0,0 +1,58 @@
+index 0c0568a..f12d50e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -105,15 +105,17 @@ macro(pyside_config option output_var)
```
```suggestion
@@ -105,15 +105,16 @@ macro(pyside_config option output_var)
```

spencerpogo commented:
I greatly cleaned up the diff thanks to reviewers. This PR is now ready for another review, but not ready for merging yet. It depends on the sub-PRs #369305 and #370105 being merged first.

ConnorNelson commented on pkgs/development/python-modules/angr/default.nix:
```diff
@@ -87,7 +95,7 @@ buildPythonPackage rec {
     sqlalchemy
     sympy
-    unicorn
+    unicorn'
```
It looks like we should now have the ability to depend on `unicorn-angr` (even if just temporarily):
https://github.com/NixOS/nixpkgs/pull/388155

fabaff committed:
python313Packages.binsync: 5.2.0 -> 5.3.0

Diff: https://github.com/binsync/binsync/compare/refs/tags/v5.2.0...v5.3.0

Changelog: https://github.com/binsync/binsync/releases/tag/v5.3.0

fabaff committed:
python313Packages.binsync: refactor

fabaff committed:
python312Packages.pyqodeng-angr: 0.0.11 -> 0.0.13

Diff: https://github.com/angr/pyqodeng/compare/refs/tags/v0.0.11...v0.0.13

Changelog: https://github.com/angr/pyqodeng/blob/refs/tags/v0.0.13/CHANGELOG.rst

fabaff committed:
angr-management: 9.2.130 -> 9.2.147

Diff: https://github.com/angr/angr-management/compare/refs/tags/v9.2.130...v9.2.147

Changelog: https://github.com/angr/angr-management/releases/tag/v9.2.147

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 360310`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 34 packages built:</summary>
  <ul>
    <li>angr-management</li>
    <li>angr-management.dist</li>
    <li>mov-cli</li>
    <li>mov-cli.dist</li>
    <li>python312Packages.binsync</li>
    <li>python312Packages.binsync.dist</li>
    <li>python312Packages.ipython-pygments-lexers</li>
    <li>python312Packages.ipython-pygments-lexers.dist</li>
    <li>python312Packages.pyqodeng-angr</li>
    <li>python312Packages.pyqodeng-angr.dist</li>
    <li>python312Packages.qtawesome</li>
    <li>python312Packages.qtawesome.dist</li>
    <li>spyder (python312Packages.spyder)</li>
    <li>python312Packages.spyder-kernels</li>
    <li>python312Packages.spyder-kernels.dist</li>
    <li>spyder.dist (python312Packages.spyder.dist)</li>
    <li>python312Packages.thefuzz</li>
    <li>python312Packages.thefuzz.dist</li>
    <li>python313Packages.binsync</li>
    <li>python313Packages.binsync.dist</li>
    <li>python313Packages.ipython-pygments-lexers</li>
    <li>python313Packages.ipython-pygments-lexers.dist</li>
    <li>python313Packages.qtawesome</li>
    <li>python313Packages.qtawesome.dist</li>
    <li>python313Packages.spyder</li>
    <li>python313Packages.spyder-kernels</li>
    <li>python313Packages.spyder-kernels.dist</li>
    <li>python313Packages.spyder.dist</li>
    <li>python313Packages.thefuzz</li>
    <li>python313Packages.thefuzz.dist</li>
    <li>rare</li>
    <li>rare.dist</li>
    <li>sublime-music</li>
    <li>sublime-music.dist</li>
  </ul>
</details>

fabaff commented:
`angr-management` works till one tries to open a file (seg fault).

fabaff commented:
Let me see what I can do about the stacktrace.

ConnorNelson commented:
> Let me see what I can do about the stacktrace.

Hey, any updates? Still haven't been able to reproduce. It seems to open programs for analysis fine.

ConnorNelson commented:
@fabaff Just following up, what can we do to get this merged?

timschumi commented on pkgs/by-name/an/angr-management/package.nix:
```diff
@@ -0,0 +1,65 @@
+        pyzmq
+        traitlets
+      ]
+      ++ angr.optional-dependencies.AngrDB
```
`AngrDB` is lowercase now.

timschumi commented on pkgs/development/python-modules/pyqodeng-angr/default.nix:
```diff
@@ -0,0 +1,59 @@
+
+  dependencies = [
+    pygments
+    future
```
`future` no longer works with `python3.13`, so this needs angr/pyqodeng#15 (either in patch form or by waiting for a release).

timschumi replied:
I just noticed that `pyqodeng-angr` got broken out into a new PR and merged, so this should be fixed in the existing package instead.

timschumi reviewed (commented):
The package works fine for me without any apparent crashes, so just marking some things that I had to do to get it to compile a year later.
